kukemc/I18n.js

```diff
@@ -0,0 +1,1239 @@
+          defaultValue: "Hello World!",
+        },
+        LANG1: {
+          type: Scratch.ArgumentType.STRING,
+          menu: "TRANSLATE_LIST",
+          defaultValue: "en",
+        },
+        LANG2: {
+          type: Scratch.ArgumentType.STRING,
+          menu: "TRANSLATE_LIST",
+          defaultValue: "zh",
+        },
+      },
+    };
+
+    const convertUnit = {
+      opcode: "convertUnit",
+      blockType: Scratch.BlockType.REPORTER,
+      text: this.formatMessage("kukeMCI18n.block.convertUnit"),
+      arguments: {
+        NUM: {
+          type: Scratch.ArgumentType.NUMBER,
+          defaultValue: 150000,
+        },
+        UNIT: {
+          type: Scratch.ArgumentType.STRING,
+          menu: "UNIT_LIST",
+          defaultValue: "cn",
+        },
+      },
+    };
+
+    const getTimestamp = {
+      opcode: "getTimestamp",
+      blockType: Scratch.BlockType.REPORTER,
+      text: this.formatMessage("kukeMCI18n.block.getTimestamp"),
+      arguments: {},
+    };
+
+    const getTimeZone = {
+      opcode: "getTimeZone",
+      blockType: Scratch.BlockType.REPORTER,
+      text: this.formatMessage("kukeMCI18n.block.getTimeZone"),
+      arguments: {},
+    };
+
+    const offsetTimeZone = {
+      opcode: "offsetTimeZone",
+      blockType: Scratch.BlockType.REPORTER,
+      text: this.formatMessage("kukeMCI18n.block.offsetTimeZone"),
+      arguments: {
+        TIME: {
+          type: Scratch.ArgumentType.NUMBER,
+          defaultValue: "1609459200",
+        },
+        ZONE1: {
+          type: Scratch.ArgumentType.STRING,
+          defaultValue: "Asia/Shanghai",
+        },
+        ZONE2: {
+          type: Scratch.ArgumentType.STRING,
+          defaultValue: "America/New_York",
+        },
+      },
+    };
+
+    const formatTimestamp = {
+      opcode: "formatTimestamp",
+      blockType: Scratch.BlockType.REPORTER,
+      text: this.formatMessage("kukeMCI18n.block.formatTimestamp"),
+      arguments: {
+        TIME: {
+          type: Scratch.ArgumentType.NUMBER,
+          defaultValue: 1609459200,
+        },
+        FORMAT: {
+          type: Scratch.ArgumentType.STRING,
+          defaultValue: "YYYY-MM-DD HH:mm:ss",
+        },
+      },
+    };
+
+    const replaceText = {
+      opcode: "replaceText",
+      blockType: Scratch.BlockType.REPORTER,
+      text: "replace text [TEXT]",
+      arguments: {
+        TEXT: {
+          type: Scratch.ArgumentType.STRING,
+          defaultValue:
+            "My name is [我的变量%scr], nickName is [i%ark], fullName is [data%ads], bio is [key%i18n]",
+        },
+      },
+    };
+
+    return {
+      id: kukemc_i18n_extensionId,
+      name: "I18n",
+      blockIconURI: kukemc_i18n_icon,
+      menuIconURI: kukemc_i18n_icon,
+      color1: "#13003a",
+      color2: "#0a001e",
+      docsURI:
+        "https://learn.ccw.site/article/99e0432c-98f2-4394-8a32-e501beee1e27",
+      blocks: [
+        "---" + this.formatMessage("kukeMCI18n.div.1"),

```

Comment:
这样并不是标准的写法，即使 CCW 支持。
p.s. 这是 Gandi，Eureka，TW 均支持的写法，具有兼容性。
```suggestion
        {
            blockType: Scratch.BlockType.LABEL,
            text: this.formatMessage("kukeMCI18n.div.1")
        }
```

Reply:
怎么又 resolved 了，我并没有看到你把其它的都改了。